exp/runtime/hooks/api/v1alpha1/aggregate.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	runtimev1 "sigs.k8s.io/cluster-api/exp/runtime/api/v1alpha1"
+)
+
+type AggregatableResponse interface {
+	runtime.Object
+
+	Aggregate(items []*ExtensionResponse) (HookResponseSummary, error)
+}
+
+type ExtensionResponse struct {
+	ExtensionName string
+	Response      runtime.Object
+	FailurePolicy *runtimev1.FailurePolicy
+	Error         error
+}
+
+type ExtensionResponseList []*ExtensionResponse
+
+type ResponseSummaryStatus string
+
+const (
+	ResponseSummaryStatusSuccess        ResponseSummaryStatus = "Success"
+	ResponseSummaryStatusFailureIgnored ResponseSummaryStatus = "FailureIgnored"
+	ResponseSummaryStatusFailure        ResponseSummaryStatus = "Failure"
+)
+
+type ExtensionResponseSummary struct {
+	ExtensionName string
+	Status        ResponseSummaryStatus
+	Error         error
+}
+
+type HookResponseSummary []ExtensionResponseSummary
+
+func aggregateExtensionResponses(response runtime.Object, responses []*ExtensionResponse) (HookResponseSummary, error) {
```

Comment:
I wonder if we should just simplify for now and drop HookResponseSummary until we know exactly if we need it and with which information (except we already do)